src/Extensions/Abstractions/OmexServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Omex.Extensions.Abstractions
+{
+	/// <summary> Extension methods for the <see cref="IServiceCollection"/> class</summary>
+	public static class OmexServiceCollectionExtensions
+	{
+		/// <summary>Add IServiceContext to ServiceCollection</summary>
+		public static IServiceCollection AddServiceContext<TServiceContext>(this IServiceCollection serviceCollection)
```

Comment:
>serviceCollection [](start = 94, length = 17)

check for nulls?

Reply:
All of the new projects in this PR are intended to use Non-Nulable reference types feature from C# 8 so if there don't have '?' after type name they can't be null